test/dataframe_builders_test.ml

```diff
@@ -0,0 +1,127 @@
+open! Core
+open Polars
+
+let%expect_test "Basic clear" =
+  let df =
+    Data_frame.create_exn
+      Series.[ int "integer" [ 1; 2; 3; 4; 5 ]; float "float" [ 4.; 5.; 6.; 7.; 8. ] ]
+  in
+  let df2 = Data_frame.clear df in
+  Data_frame.to_string_hum df |> print_endline;
+  Data_frame.to_string_hum df2 |> print_endline;
+  [%expect
+    {|
+    shape: (5, 2)
+    ┌─────────┬───────┐
+    │ integer ┆ float │
+    │ ---     ┆ ---   │
+    │ i64     ┆ f64   │
+    ╞═════════╪═══════╡
+    │ 1       ┆ 4.0   │
+    │ 2       ┆ 5.0   │
+    │ 3       ┆ 6.0   │
+    │ 4       ┆ 7.0   │
+    │ 5       ┆ 8.0   │
+    └─────────┴───────┘
+    shape: (0, 2)
+    ┌─────────┬───────┐
+    │ integer ┆ float │
+    │ ---     ┆ ---   │
+    │ i64     ┆ f64   │
+    ╞═════════╪═══════╡
+    └─────────┴───────┘ |}]
+;;
+
+let%expect_test "Basic vstack" =
+  let df =
+    Data_frame.create_exn
+      Series.[ int "integer" [ 1; 2; 3; 4; 5 ]; float "float" [ 4.; 5.; 6.; 7.; 8. ] ]
+  in
+  let df2 =
+    Data_frame.create_exn
+      Series.[ int "integer" [ 1; 2; 3; 4; 5 ]; float "float" [ 4.; 5.; 6.; 7.; 8. ] ]
+  in
+  Data_frame.vstack_exn df ~other:df2;
+  Data_frame.to_string_hum df |> print_endline;
+  [%expect
+    {|
+    shape: (10, 2)
+    ┌─────────┬───────┐
+    │ integer ┆ float │
+    │ ---     ┆ ---   │
+    │ i64     ┆ f64   │
+    ╞═════════╪═══════╡
+    │ 1       ┆ 4.0   │
+    │ 2       ┆ 5.0   │
+    │ 3       ┆ 6.0   │
+    │ 4       ┆ 7.0   │
+    │ …       ┆ …     │
+    │ 2       ┆ 5.0   │
+    │ 3       ┆ 6.0   │
+    │ 4       ┆ 7.0   │
+    │ 5       ┆ 8.0   │
+    └─────────┴───────┘ |}]
+;;
+
+let build_a_dataframe_usage n =
+  let df = Data_frame.create_exn Series.[ int "integer" []; float "float" [] ] in
+  let single_row =
+    Data_frame.create_exn Series.[ int "integer" [ 1 ]; float "float" [ 4. ] ]
+  in
+  List.init n ~f:Fn.id
+  |> List.iter ~f:(fun i ->
+    Data_frame.vstack_exn df ~other:single_row;
+    if i % 100 = 0
+    then (
+      let (_ : Data_frame.t) =
+        Sql_context.rust_sql_context_execute_with_data_frames_exn
+          ~data_frames_with_names:[ df, "data" ]
+          ~query:"select * from data limit 1"
+      in
+      ()));
+  Data_frame.as_single_chunk_par df;
+  let count =
+    Sql_context.rust_sql_context_execute_with_data_frames_exn
+      ~data_frames_with_names:[ df, "data" ]
+      ~query:"select count(*), max(integer) from data"
+  in
+  Data_frame.to_string_hum count |> print_string
+;;
+
+let%bench "build and periodically query a dataframe -- 100_000 iters" =
+  build_a_dataframe_usage 100_000
+;;
```

Comment:
ppx_bench is awesome, but every time I try to use it, I forget how to run it. Could you maybe add a small mention of how to run it in https://github.com/mt-caret/polars-ocaml/blob/main/HACKING.md?

Reply:
I'm actually not sure either. Normally, I'm used to jenga outputting inline_test_runner and inline_benchmarks_runner executables.  I've just been testing in jane and I looked in dune/test to see whether I could construct the same executables via dune but didn't get very far. Have you gotten this to work before?

Reply:
Hmm, ack; I don't remember ever trying ppx_bench outside of work; I'll try playing around with dune to see if I can get it to work.

Reply:
Done, based off the example here: https://github.com/ocaml/dune/tree/776813d59bf22cb1ce35ed2299b4b8c0e30fae21/bench/micro. I've also changed the benchmark to ignore the resulting dataframe instead of printing it out since I'm pretty sure we don't want to include that in our benchmark.